www/caddy/src/opnsense/mvc/app/views/OPNsense/Caddy/reverse_proxy.volt

```diff
@@ -143,10 +163,72 @@
 
         // Initialize the service control UI for 'caddy'
         updateServiceControlUI('caddy');
+        
+        // Filter function for domains
+        function loadDomainFilters() {
+            $.ajax({
+                url: '/api/caddy/ReverseProxy/getAllReverseDomains', // custom API endpoint to get uuid and domainport combinations
+                type: 'GET',
+                dataType: 'json',
+                success: function(data) {
+                    var select = $('#reverseFilter');
+                    select.empty(); // Clear current options
+                    if (data && data.rows) {
+                        data.rows.forEach(function(item) {
+                            select.append($('<option>').val(item.id).text(item.domainPort));
+                        });
+                    }
+                    select.selectpicker('refresh'); // Refresh selectpicker to update the UI
+                },
+                error: function() {
+                    $('#reverseFilter').html('<option value="">Failed to load data</option>').selectpicker('refresh');
+                }
+            });
+        }
+        loadDomainFilters();
+        
+        // Reload Bootgrid on filter change
+        $('#reverseFilter').on('changed.bs.select', function() {
+            $("#reverseProxyGrid").bootgrid("reload");
+            $("#reverseSubdomainGrid").bootgrid("reload");
+            $("#reverseHandleGrid").bootgrid("reload");
+        });
+        
+        // Control the visibility of selectpicker for filter by domain
+        function toggleSelectPicker(tab) {
+            if (tab === 'handlesTab' || tab === 'domainsTab') {
+                $('.common-filter').show();
+            } else {
+                $('.common-filter').hide();
+            }
+        }
+
+        // Initialize visibility based on the active tab on page load
+        var activeTab = $('#maintabs .active a').attr('href').replace('#', '');
+        toggleSelectPicker(activeTab);
+
+        // Change event when switching tabs
+        $('#maintabs a').on('click', function (e) {
+            var currentTab = $(this).attr('href').replace('#', '');
+            toggleSelectPicker(currentTab);
+        });
 
     });
 </script>
 
+<style>
+    .common-filter {
+        text-align: right;
+        margin-top: 20px;
+        margin-right: 5px;
+        padding: 0 15px;  // Align with the tables
+    }
+
+    .selectpicker {
+        width: 100% !important;  // Ensure it fills the container
```

Comment:
this restyling is pretty broad and excessiv (!important)

you should narrow this to a second class or setting id because it may have unintended side effects

Reply:
Okay I restricted it to an ID.